src/nodes.coffee

```diff
@@ -502,8 +502,6 @@ exports.Block = class Block extends Base
         # We want to compile this and ignore the result.
         node.compileToFragments o
         continue
-
-      node = node.unwrapAll()
```

Comment:
@connec Do you think there’s any concern about me removing this line? I removed it in order to not necessarily always optimize away redundant parentheses (which is what we’re dealing with in this case).

Reply:
@GeoffreyBooth why would we not want to optimize away redundant parentheses? Otherwise nothing jumps out at me!

Reply:
Because of this particular case. When parentheses wrap an `IdentifierLiteral` that has a comment, we want those parentheses to be output because Flow requires them. See https://github.com/jashkenas/coffeescript/issues/4706#issuecomment-330533014.

I’m assuming that removing this optimization doesn’t _break_ anything else, though, right?